tests/unit/aiplatform/test_pipeline_jobs.py

```diff
@@ -0,0 +1,193 @@
+import json
+
+from unittest import mock
+from importlib import reload
+from unittest.mock import patch
+
+from google.auth import credentials as auth_credentials
+
+from google.cloud import aiplatform
+from google.cloud import storage
+
+from google.cloud.aiplatform import pipeline_jobs
+from google.cloud.aiplatform import initializer
+from google.cloud.aiplatform.utils import json_utils
+
+from google.cloud.aiplatform_v1beta1.services.pipeline_service import (
+    client as pipeline_service_client_v1beta1,
+)
+
+from google.cloud.aiplatform_v1beta1.types import (
+    pipeline_job as gca_pipeline_job_v1beta1,
+    pipeline_state as gca_pipeline_state_v1beta1,
+)
+
+_TEST_PROJECT = "test-project"
+_TEST_LOCATION = "us-central1"
+_TEST_PIPELINE_JOB_ID = "sample-test-pipeline-202111111"
+_TEST_GCS_BUCKET_NAME = "my-bucket"
+_TEST_CREDENTIALS = mock.Mock(spec=auth_credentials.AnonymousCredentials())
+_TEST_SERVICE_ACCOUNT = "abcde@my-project.iam.gserviceaccount.com"
+
+_TEST_JOB_SPEC_PATH = f"gs://{_TEST_GCS_BUCKET_NAME}/job_spec.json"
+_TEST_PARENT = f"projects/{_TEST_PROJECT}/locations/{_TEST_LOCATION}"
+_TEST_NETWORK = f"projects/{_TEST_PROJECT}/global/networks/{_TEST_PIPELINE_JOB_ID}"
+
+_TEST_PIPELINE_JOB_NAME = f"projects/{_TEST_PROJECT}/locations/{_TEST_LOCATION}/pipelineJobs/{_TEST_PIPELINE_JOB_ID}"
+
+_TEST_MACHINE_TYPE = "n1-standard-4"
+_TEST_ACCELERATOR_TYPE = "NVIDIA_TESLA_P100"
+_TEST_ACCELERATOR_COUNT = 2
+_TEST_STARTING_REPLICA_COUNT = 2
+_TEST_MAX_REPLICA_COUNT = 12
+
+_TEST_PIPELINE_PARAMETER_VALUES = {"name_param": "hello"}
+_TEST_PIPELINE_JOB_SPEC = {
+    "runtimeConfig": {},
+    "pipelineSpec": {
+        "pipelineInfo": {"name": "my-pipeline"},
+        "root": {
+            "dag": {"tasks": {}},
+            "inputDefinitions": {"parameters": {"name_param": {"type": "STRING"}}},
+        },
+        "components": {},
+    },
+}
+
+_TEST_PIPELINE_GET_METHOD_NAME = "get_fake_pipeline_job"
+_TEST_PIPELINE_LIST_METHOD_NAME = "list_fake_pipeline_jobs"
+_TEST_PIPELINE_CANCEL_METHOD_NAME = "cancel_fake_pipeline_job"
+_TEST_PIPELINE_DELETE_METHOD_NAME = "delete_fake_pipeline_job"
+_TEST_PIPELINE_RESOURCE_NAME = (
+    f"{_TEST_PARENT}/fakePipelineJobs/{_TEST_PIPELINE_JOB_ID}"
+)
+
+
+@pytest.fixture
+def mock_pipeline_service_create():
+    with mock.patch.object(
+        pipeline_service_client_v1beta1.PipelineServiceClient, "create_pipeline_job"
+    ) as mock_create_pipeline_job:
+        mock_create_pipeline_job.return_value = gca_pipeline_job_v1beta1.PipelineJob(
+            name=_TEST_PIPELINE_JOB_NAME,
+            state=gca_pipeline_state_v1beta1.PipelineState.PIPELINE_STATE_SUCCEEDED,
+        )
+        yield mock_create_pipeline_job
+
+
+@pytest.fixture
+def mock_pipeline_service_get():
+    with mock.patch.object(
+        pipeline_service_client_v1beta1.PipelineServiceClient, "get_pipeline_job"
+    ) as mock_get_pipeline_job:
+        mock_get_pipeline_job.return_value = gca_pipeline_job_v1beta1.PipelineJob(
```

Comment:
I updated the mock_pipeline_service_get in test_training_jobs.py here iteratively return multiple payloads to reproduce the polling of a running pipeline using `side_effect` instead of `return_value`: https://github.com/googleapis/python-aiplatform/blob/master/tests/unit/aiplatform/test_training_jobs.py#L457

I think it would get good to have the same pattern here as it will test the inner block of the wait loop more thoroughly.